implants/lib/eldritch/src/file/moveto_impl.rs

```diff
@@ -6,97 +6,93 @@ pub fn moveto(old: String, new: String) -> Result<()> {
     // If path is a dir delete it.
     // This will help unify behavior across systems.
     // https://doc.rust-lang.org/std/fs/fn.rename.html#platform-specific-behavior
-    if Path::new(&new.clone()).is_dir() {
+    if Path::new(&new).is_dir() {
         fs::remove_dir_all(new.clone())?;
-    } else if Path::new(&new.clone()).is_file() {
+    } else if Path::new(&new).is_file() {
         fs::remove_file(new.clone())?;
     }
 
     fs::rename(old, new)?;
     Ok(())
 }
 
-
 #[cfg(test)]
 mod tests {
     use super::*;
-    use tempfile::{NamedTempFile,tempdir};
-    use std::path::Path;
     use std::fs::File;
+    use std::path::Path;
+    use tempfile::{tempdir, NamedTempFile};
 
     #[test]
-    fn test_moveto_dir() -> anyhow::Result<()>{
+    fn test_moveto_dir() -> anyhow::Result<()> {
         let tmp_dir = tempdir()?;
-        let path_old = String::from(tmp_dir.path().to_str().unwrap()).clone();
+        let path_old = String::from(tmp_dir.path().to_str().unwrap());
         let tmp_dir_new = tempdir()?;
-        let path_new = String::from(tmp_dir_new.path().to_str().unwrap()).clone();
+        let path_new = String::from(tmp_dir_new.path().to_str().unwrap());
         tmp_dir_new.close()?;
 
-
         moveto(path_old.clone(), path_new.clone())?;
 
-        assert_eq!(Path::new(&path_old).is_dir(), false);
-        assert_eq!(Path::new(&path_new).is_dir(), true);
+        assert!(!Path::new(&path_old).is_dir());
```

Comment:
Lets switch this to .exists() instead of .is_file() while were at it. I think that's more clear this is checking the file no longer exists.